artifactory_cleanup/artifactorycleanup.py

```diff
@@ -56,9 +59,12 @@ def cleanup(self, block_ctx_mgr, test_ctx_mgr) -> Iterator[CleanupSummary]:
                 print(f"Found {len(artifacts_to_remove)} artifacts AFTER filtering")
 
                 # Delete artifacts
-                for artifact in artifacts_to_remove:
-                    with test_ctx_mgr(get_name_for_ci(artifact)):
-                        policy.delete(artifact, destroy=self.destroy)
+                with ThreadPoolExecutor(max_workers=int(self.worker_count)) as executor:
+                    for artifact in artifacts_to_remove:
+                        with test_ctx_mgr(get_name_for_ci(artifact)):
+                            executor.submit(
+                                policy.delete, artifact, destroy=self.destroy, ignore_not_found=self.ignore_not_found
+                            )
```

Comment:
Let's move `with` context manager inside the submit, so we'll submit everything incide "for" cycle

```python
def _task(args, kwargs):
    with test_ctx_mgr(get_name_for_ci(artifact)):
        policy.delete(artifact, destroy=self.destroy)
executor.submit(self._task, args, kwargs)
```